agent/src/storage/bucket.rs

```diff
@@ -0,0 +1,169 @@
+use aws_sdk_s3 as s3;
+use s3::error::ProvideErrorMetadata;
+use s3::primitives::ByteStream;
+use ssri::Integrity;
+use utils::errors::{ServalError, ServalResult};
+use utils::structs::Manifest;
+
+#[derive(Debug, Clone)]
+pub struct S3Storage {
+    client: s3::Client,
+    bucket: String,
+}
+
+impl S3Storage {
+    pub fn new(bucket_name: &str, config: aws_config::SdkConfig) -> ServalResult<Self> {
+        let client = s3::Client::new(&config);
```

Comment:
Asking because I'm still on that learning treadmill: I am surprised to see `config` passed in by reference, rather than handing ownership over. Is that surprising to you, too?